File: tools/changes.py
from rich import print

import subprocess


def main():
    previous = read_version()
    print(f"Changes since {previous}:")
    print(changes(previous))


def read_version() -> str:
    with open("version", "r") as f:
        return f.read()


def changes(previous: str) -> str:
    command = ["git", "hist", previous + "..HEAD"]

    result = subprocess.run(command, text=True, capture_output=True)

    if result.returncode == 0:
        return result.stdout
    else:
        return result.stderr


if __name__ == "__main__":
    main()

File: archive/devyctl/src/tools/changes.rs
use lib::Git;

pub fn changes() {
    let version = std::fs::read_to_string("./version").expect("Could not read version file");

    let git = Git::new("/opt/homebrew/bin/git").expect("Could not find git");
    let hist = git
        .hist(".", &version, "HEAD")
        .expect("Could not get git history");

    println!("{}", hist);
}
